modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -174,5 +188,23 @@ def enum_zones
       logger.debug "Enumerated authoritative dns zones: #{zones}"
       zones
     end
+
+    def enum_records(zone_name, node_name, type)
+      records = []
+      response = execute "/EnumRecords #{zone_name} #{node_name}. /Type #{type}", "EnumRecords", true
+      response.each do |line|
+        line.chomp!
+        logger.debug "Extracting record from dnscmd output '#{line}'"
+        matches = /^@\s+(?<ttl>\d+)\s+(?<type>#{type})\s+(?<record>\S+)$/.match(line)
```

Comment:
I don't think you need all these groups -- only the record bit is being used... I think something like 
```/^@\s+\d+\s+[[:upper:]]+\s+(?<record>\S+)/ ``` should do the trick. A nitpick: you could use ```=~``` to match the response; ```record``` variable will either contain the match or nil then.